### PR TITLE
refactor(angular): refactored angular related imports generation

### PR DESCRIPTION
**Pull request by g-cheishvili**
## Description

- added `populatedWithAngularImports` and moved adding Angular related imports to be BEFORE `json.post` call.
- added one helper for adding an import to `MitosisComponent`. I do not know if something like it existed, could not find it, so added it. if there is any, please refer me to it.

## Reasoning
I added these because otherwise in `json.post` you can not see which imports are gonna end up in file completely. Also, output code's structure is now better because it follows a widely accepted pattern:
- imports
- types
- anything else
- class declaration

### CODE REVIEW

File: packages/core/src/helpers/add-import.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { MitosisComponent, MitosisImport } from '@builder.io/mitosis';
+
+/**
+ * Add an import to a component, if it doesn't already exist.
+ * Returns modified MitosisComponent, which is NOT a new object
+ * @param json The component to add the import to
+ * @param theImport The import to add
+ */
+export function addImportToMitosisComponent(
```
Comment (samijaber) [inline packages/core/src/helpers/add-import.ts:9]: We don't have this yet. Thanks for putting it together!

**Review by samijaber — APPROVED** [review]

File: packages/core/src/helpers/add-import.ts (hunk 1 above)
Comment (g-cheishvili) [reply packages/core/src/helpers/add-import.ts:9]: Aand I need to update snapshots here as well

**Comment by g-cheishvili** [comment]
@samijaber fixed everything. Take a look

**Comment by samijaber** [comment]
@g-cheishvili I believe this is still pending updating snapshots?

**Comment by PatrickJS** [comment]
@g-cheishvili any updates on this?

**Pull request closed without merging**